nsxt/resource_nsxt_lb_cookie_persistence_profile.go

```diff
@@ -0,0 +1,275 @@
+/* Copyright © 2018 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	api "github.com/vmware/go-vmware-nsxt"
+	"github.com/vmware/go-vmware-nsxt/loadbalancer"
+	"log"
+	"net/http"
+)
+
+var cookieModeTypes = []string{"INSERT", "PREFIX", "REWRITE"}
+var cookieExpiryTypes = []string{"SESSION_COOKIE_TIME", "PERSISTENCE_COOKIE_TIME"}
+
+func resourceNsxtLbCookiePersistenceProfile() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtLbCookiePersistenceProfileCreate,
+		Read:   resourceNsxtLbCookiePersistenceProfileRead,
+		Update: resourceNsxtLbCookiePersistenceProfileUpdate,
+		Delete: resourceNsxtLbCookiePersistenceProfileDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"revision": getRevisionSchema(),
+			"description": &schema.Schema{
+				Type:        schema.TypeString,
+				Description: "Description of this resource",
+				Optional:    true,
+			},
+			"display_name": &schema.Schema{
+				Type:        schema.TypeString,
+				Description: "The display name of this resource. Defaults to ID if not set",
+				Optional:    true,
+				Computed:    true,
+			},
+			"tag": getTagsSchema(),
+			"cookie_mode": &schema.Schema{
+				Type:         schema.TypeString,
+				Description:  "The cookie persistence mode",
+				Optional:     true,
+				Default:      "INSERT",
+				ValidateFunc: validation.StringInSlice(cookieModeTypes, false),
+			},
+			"cookie_name": &schema.Schema{
+				Type:        schema.TypeString,
+				Description: "The name of the cookie",
+				Required:    true,
+			},
+			"persistence_shared": &schema.Schema{
+				Type:        schema.TypeBool,
+				Description: "A boolean flag which reflects whether the cookie persistence is private or shared",
+				Optional:    true,
+				Default:     false,
+			},
+			"cookie_fallback": &schema.Schema{
+				Type:        schema.TypeBool,
+				Description: "A boolean flag which reflects whether once the server points by this cookie is down, a new server is selected, or the requests will be rejected",
+				Optional:    true,
+				Default:     true,
+			},
+			"cookie_garble": &schema.Schema{
+				Type:        schema.TypeBool,
+				Description: "A boolean flag which reflects whether the cookie value (server IP and port) would be encrypted or in plain text",
+				Optional:    true,
+				Default:     true,
+			},
+			"cookie_domain": &schema.Schema{
```

Comment:
I suggest to group all insert params into sub-struct

Reply:
ok. Done.